sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/HiveThriftServer2Listener.scala

```diff
@@ -180,10 +180,12 @@ private[thriftserver] class HiveThriftServer2Listener(
   }
 
   private def onOperationClosed(e: SparkListenerThriftServerOperationClosed): Unit = {
-    executionList.get(e.id).closeTimestamp = e.closeTime
-    executionList.get(e.id).state = ExecutionState.CLOSED
-    updateStoreWithTriggerEnabled(executionList.get(e.id))
-    executionList.remove(e.id)
+    if (executionList.contains(e.id)) {
```

Comment:
I think we can add a comment here to explain? This issue can happen only on cancel operations right?

Reply:
> I think we can add a comment here to explain? This issue can happen only on cancel operations right?

Some operation `FAILED|CANCELED` and remove statementId before call `closeOperation `
in current master brach.

Reply:
Yeah, we could add comment like `Operations that were CANCELED or FAILED may have already been removed.`

But now it doesn't happen because we don't trim LiveExecutionData?

Reply:
Yes, It seems this issue doesn't happen as we aren't removing `LiveExecutionData`.  @AngersZhuuuu Could you make `spark.sql.thriftserver.ui.retainedStatements=1` and try to reproduce the same?

Reply:
> Yes, It seems this issue doesn't happen as we aren't removing `LiveExecutionData`. @AngersZhuuuu Could you make `spark.sql.thriftserver.ui.retainedStatements=1` and try to reproduce the same?

So complicated now... executionList won't remove, and then `updateStoreWithTriggerEnabled ` may cause some problem.  I will test it .

Reply:
I don't think, `updateStoreWithTriggerEnabled` cause a problem, as it write a new entry to the kvstore.

Reply:
> I don't think, `updateStoreWithTriggerEnabled` cause a problem, as it write a new entry to the kvstore, like in the historyserver case.

seems we don't need to fix this now.

Reply:
> seems we don't need to fix this now.

Yes, Right. But still in branch-2.4 this issue can happen right?

Reply:
> > seems we don't need to fix this now.
> 
> Yes, Right. But still in branch-2.4 this issue can happen right?

Don't need, since i spark-2.4, didn't add `onStatementClosed & onStatementCanceled`